aiohttp/web.py

```diff
@@ -263,6 +263,24 @@ def __repr__(self):
         return "<Application 0x{:x}>".format(id(self))
 
 
+############################################################
+# Helper functions
+############################################################
+
+def named_url(app_or_request, name, **kwargs):
+    app = getattr(app_or_request, 'app', app_or_request)
```

Comment:
Which is precisely why I wanted to make it a method on those two classes. :smile: 
But no worries, I'm adding the check (finally found the time to work on this).